piecedirectory/doctor_test.go

```diff
@@ -44,56 +64,166 @@ func testPieceDoctor(ctx context.Context, t *testing.T, bdsvc *svc.Service, port
 	require.NoError(t, err)
 	defer cl.Close(ctx)
 
+	//t.Run("next pieces", func(t *testing.T) {
+	//testNextPieces(ctx, t, cl, pieceCheckPeriod)
+	//})
+
+	t.Run("next pieces pagination", func(t *testing.T) {
+		testNextPiecesPagination(ctx, t, cl, pieceCheckPeriod)
+	})
+
 	t.Run("check pieces", func(t *testing.T) {
 		testCheckPieces(ctx, t, cl)
 	})
-
-	// TODO: uncomment once the leveldb implementation is complete
-	//t.Run("next pieces", func(t *testing.T) {
-	//	testNextPieces(ctx, t, cl, pieceCheckPeriod)
-	//})
 }
 
 // Verify that after a new piece is added
 // - NextPiecesToCheck immediately returns the piece
 // - NextPiecesToCheck returns the piece every pieceCheckPeriod
```

Comment:
@dirkmc I think it is hard to implement this in LevelDB, unless we add book-keeping for any new piece, which I'd like to avoid.

New pieces are added in a sorted table in LevelDB. What I've implemented is in-memory offset tracking and the doctor going batch by batch over all entries in the table. This is slightly different to the Couchbase implementation, but I think it is good enough. WDYT?

Reply:
Yes I think it's fine for them to have different implementations 👍